### DISCUSSION TITLE
Any way to get motion events with blocked Internet access?

### DISCUSSION THREAD

**Discussion by brianjmurrell** [JurajNyiri/HomeAssistant-Tapo-Control#663] [Q&A]
I block all outgoing Internet access from my C200 to prevent any hacking of TP-Link's Cloud control-centre resulting in hacking into my camera's video feed.  But that also seems to block the camera's motion detection capabilities.

Is there any way to get motion events purely locally, without enabling Internet access on these cameras?

I understand that ONVIF is supposed to be able to [handle motion events](https://www.google.com/search?q=onvif+motion+events&oq=ONVIF+motion).  Is this a possible work-around for these camera's needing Internet access to trigger motion events?

**Comment by JurajNyiri** [comment]
Hi yes integration supports this locally (only basic movement sensor). See faq in readme if it is not showing up.

**Reply by brianjmurrell** [reply]
I have read the FAQ and I do get motion sensor events from the integration, but only when I unblock the camera from the Internet.  As soon as I block it again, no more motion events in HA.

**Reply by JurajNyiri** [reply]
The onvif works locally, make sure HA can access it locally properly, it needs to access onvif 2020 port and camera needs to be able to reach HA as well.

**Reply by brianjmurrell** [reply]
Yes, I know HA can reach ONVIF on the camera as I used the ONVIF integration to make sure it was there and working.

**Reply by brianjmurrell** [reply]
Oh, interesting.  It does seem to have started working:

![image](https://github.com/user-attachments/assets/955ce895-e077-4138-b01a-22cbca308167)

Not sure how accurate it's being though.  I doubt there really has been motion in that room for a full hour now.  Maybe I suppose.  Anyway now that there is at least indication that it's working, I'll play with it a bit more.
